### PR TITLE
Feature/lga 2182 add pre-commit

### PR DESCRIPTION
**Pull request by psweeting1**
Adding pre-commit which contains flake8 and black configs

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+repos:
+-   repo: https://github.com/ambv/black
+    rev: 18.9b0
```
Comment (mtac50) [inline .pre-commit-config.yaml:3]: I'm guessing this is just to align with the other repos but why are we using a pre-release version of black? Does pre-commit work with the latest version (22.10.0)?

**Review by mtac50 — COMMENTED** [review]
Always feels good to see well formatted code! Just the one query about the version of black we're on, but other than that its good to go :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (psweeting1) [reply .pre-commit-config.yaml]: Version of flask8 is not 5.0.4, version is black is 22.8.0 and pre-commit is now version 2.20.0. These are all the latest versions.

**Review by Pheonnexx — APPROVED** [review]
LGTM 👍

**Pull request merged**